daemon/internal/newrelic/integration/test.go

```diff
@@ -285,10 +285,15 @@ func ScrubHost(in []byte) []byte {
 	return re.ReplaceAll(in, []byte("__HOST__"))
 }
 
-func SubEnvVars(in []byte) []byte {
+func (t *Test) subEnvVars(in []byte) []byte {
 	re := regexp.MustCompile("ENV\\[.*?\\]")
 	return re.ReplaceAllFunc(in, func(match []byte) []byte {
-		return []byte(os.Getenv(string(match[4 : len(match)-1])))
+		k := string(match[4 : len(match)-1])
+		v, present := t.Env[k]
+		if !present {
+			v = os.Getenv(k)
+		}
+		return []byte(v)
```

Comment:
What happens if the key is not in in or the process environment variables? Does this return nil?

Reply:
If `t.Env` doesn't have the value assigned to `k`, and `k` is not defined in the environment, `v` will be empty - `""`.

More info at: https://go.dev/ref/spec#Index_expressions, https://pkg.go.dev/os#Getenv, https://go.dev/blog/maps

Reply:
I have added a test to illustrate this: 5a8e3f4046fe427d329e49ac3faddc9da34b8e6c.